internal/service/lakeformation/opt_in.go

```diff
@@ -0,0 +1,1133 @@
+						stringplanmodifier.UseStateForUnknown(),
+					},
+				},
+			},
+			Blocks: map[string]schema.Block{
+				"column_wildcard": schema.ListNestedBlock{
+					CustomType: fwtypes.NewListNestedObjectTypeOf[columnWildcardData](ctx),
+					Validators: []validator.List{
+						listvalidator.SizeAtMost(1),
+						listvalidator.AtLeastOneOf(
+							path.MatchRelative().AtParent().AtName("column_names"),
+							path.MatchRelative().AtParent().AtName("column_wildcard"),
+						),
+					},
+					PlanModifiers: []planmodifier.List{
+						listplanmodifier.RequiresReplace(),
+					},
+					NestedObject: schema.NestedBlockObject{
+						Attributes: map[string]schema.Attribute{
+							"excluded_column_names": schema.SetAttribute{
+								CustomType: fwtypes.SetOfStringType,
+								Optional:   true,
+								PlanModifiers: []planmodifier.Set{
+									setplanmodifier.RequiresReplace(),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"last_updated_by": schema.StringAttribute{
+				Computed: true,
+			},
+			"last_modified": schema.StringAttribute{
+				CustomType: timetypes.RFC3339Type{},
+				Computed:   true,
+			},
+		},
+		Blocks: map[string]schema.Block{
+			names.AttrCondition: schema.ListNestedBlock{
+				CustomType: fwtypes.NewListNestedObjectTypeOf[Condition](ctx),
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						names.AttrExpression: schema.StringAttribute{
+							Computed: true,
+						},
+					},
+				},
+			},
+			names.AttrPrincipal: schema.ListNestedBlock{
+				CustomType: fwtypes.NewListNestedObjectTypeOf[DataLakePrincipal](ctx),
+				Validators: []validator.List{
+					listvalidator.SizeAtLeast(1),
+				},
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"data_lake_principal_identifier": schema.StringAttribute{
+							Required: true,
+						},
+					},
+				},
+			},
+			"resource_data": schema.ListNestedBlock{
+				CustomType: fwtypes.NewListNestedObjectTypeOf[ResourceData](ctx),
+				NestedObject: schema.NestedBlockObject{
+					Blocks: map[string]schema.Block{
+						"catalog":            catalogLNB,
+						names.AttrDatabase:   databaseLNB,
+						"data_cells_filter":  dataCellsFilterLNB,
+						"data_location":      dataLocationLNB,
+						"lf_tag":             lfTagLNB,
+						"lf_tag_expression":  lftagExpressionLNB,
+						"lf_tag_policy":      lfTagPolicyLNB,
+						"table":              tableLNB,
+						"table_with_columns": tableWCLNB,
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *resourceOptIn) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	conn := r.Meta().LakeFormationClient(ctx)
+
+	var plan resourceOptInData
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	in := lakeformation.CreateLakeFormationOptInInput{}
```

Comment:
Calling it an "opt in" blurs the line with a verb but the API is using it as a noun so it seems fine to me.

Reply:
Oh sorry, the name is fine. I was just mentioning that some folks might have issues with "opt in" because it is a verb and a noun. However, since AWS is using it as a noun, such as "create an opt in", it seems doubly fine to me.